v2/googlecloud-to-googlecloud/src/test/java/com/google/cloud/teleport/v2/templates/bigtablechangestreamstogcs/BigtableChangeStreamsToGcsIT.java

```diff
@@ -0,0 +1,539 @@
+                .addParameter("windowDuration", "10s")
+                .addParameter("gcsOutputDirectory", outputPath)
+                .addParameter("schemaOutputFormat", "BIGTABLE_ROW"));
+
+    assertThatPipeline(launchInfo).isRunning();
+
+    String rowkey = UUID.randomUUID().toString();
+    String column = UUID.randomUUID().toString();
+    String value = UUID.randomUUID().toString();
+
+    long nowMillis = System.currentTimeMillis();
+    long timestampMicros = nowMillis * 1000;
+
+    RowMutation rowMutation =
+        RowMutation.create(srcTable, rowkey)
+            .setCell(SOURCE_COLUMN_FAMILY, column, timestampMicros, value);
+
+    ChangelogEntry expected = new ChangelogEntry();
+
+    expected.setRowKey(bb(rowkey));
+    expected.setTimestamp(timestampMicros);
+    expected.setCommitTimestamp(nowMillis - 10000); // clock skew tolerance
+    expected.setLowWatermark(0);
+    expected.setColumn(bb(column));
+    expected.setValue(bb(value));
+    expected.setColumnFamily(SOURCE_COLUMN_FAMILY);
+    expected.setModType(ModType.SET_CELL);
+    expected.setIsGC(false);
+
+    bigtableResourceManager.write(rowMutation);
+
+    validateGCSContents(launchInfo, "output", new LookForBigtableRowAvroRecord(expected));
+  }
+
+  @Test
+  public void testSingleMutationBigtableRowAvroWithStartTimeE2E() throws Exception {
+    LaunchInfo launchInfo =
+        launchTemplate(
+            LaunchConfig.builder(testName, specPath)
+                .addParameter("bigtableReadTableId", srcTable)
+                .addParameter("bigtableReadInstanceId", bigtableResourceManager.getInstanceId())
+                .addParameter("bigtableChangeStreamAppProfile", appProfileId)
+                .addParameter("outputFileFormat", "AVRO")
+                .addParameter("windowDuration", "10s")
+                .addParameter("gcsOutputDirectory", outputPath)
+                .addParameter("schemaOutputFormat", "BIGTABLE_ROW"));
+
+    assertThatPipeline(launchInfo).isRunning();
+
+    String rowkey = UUID.randomUUID().toString();
+    String column = UUID.randomUUID().toString();
+    String value = UUID.randomUUID().toString();
+
+    long nowMillis = System.currentTimeMillis();
+    long timestampMicros = nowMillis * 1000;
+
+    RowMutation rowMutation =
+        RowMutation.create(srcTable, rowkey)
+            .setCell(SOURCE_COLUMN_FAMILY, column, timestampMicros, value);
+
+    ChangelogEntry expected = new ChangelogEntry();
+
+    expected.setRowKey(bb(rowkey));
+    expected.setTimestamp(timestampMicros);
+    expected.setCommitTimestamp(timestampMicros - 10000000); // clock skew tolerance
+    expected.setLowWatermark(0);
+    expected.setColumn(bb(column));
+    expected.setValue(bb(value));
+    expected.setColumnFamily(SOURCE_COLUMN_FAMILY);
+    expected.setModType(ModType.SET_CELL);
+    expected.setIsGC(false);
+
+    bigtableResourceManager.write(rowMutation);
+
+    Thread.sleep(10000);
```

Comment:
nit: maybe we don't need such long sleeps

Reply:
These sleeps are tiny compared to everything else this test does (Dataflow jobs starts etc.)